File: app/src/main/java/com/elegion/android/template/data/remote/response/LoginResponse.java
package com.elegion.android.template.data.remote.response;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * @author mikhail.barannikov on 04.08.2017
 */
public class LoginResponse {
    @SerializedName("id")
    @Expose
    private int mId;

    @SerializedName("url")
    @Expose
    private String mUrl;

    @SerializedName("app")
    @Expose
    private App mApp;

    @SerializedName("token")
    @Expose
    private String mToken;

    @SerializedName("hashed_token")
    @Expose
    private String mHashedToken;

    @SerializedName("token_last_eight")
    @Expose
    private String mTokenLastEight;

    @SerializedName("note")
    @Expose
    private String mNote;

    @SerializedName("note_url")
    @Expose
    private String mNoteUrl;

    @SerializedName("created_at")
    @Expose
    private String mCreatedAt;

    @SerializedName("updated_at")
    @Expose
    private String mUpdatedAt;

    @SerializedName("scopes")
    @Expose
    private List<String> mScopes;

    @SerializedName("fingerprint")
    @Expose
    private String mFingerprintUrl;

    public String getToken() {
        return mToken;
    }

    public LoginResponse(String token) {
        mToken = token;
    }

    private static class App {
        @SerializedName("name")
        @Expose
        private String mName;

        @SerializedName("url")
        @Expose
        private String mUrl;

        @SerializedName("client_id")
        @Expose
        private String mClientId;
    }
}

File: app/src/main/java/com/elegion/android/template/data/remote/rest/response/LoginResponse.kt
package com.elegion.android.template.data.remote.rest.response

import com.google.gson.annotations.SerializedName

class LoginResponse(
        @SerializedName("token")
        var token: String
) {
    @SerializedName("id")
    private var mId: Int = 0

    @SerializedName("url")
    private var mUrl: String? = null

    @SerializedName("app")
    private var mApp: App? = null

    @SerializedName("hashed_token")
    private var mHashedToken: String? = null

    @SerializedName("token_last_eight")
    private var mTokenLastEight: String? = null

    @SerializedName("note")
    private var mNote: String? = null

    @SerializedName("note_url")
    private var mNoteUrl: String? = null

    @SerializedName("created_at")
    private var mCreatedAt: String? = null

    @SerializedName("updated_at")
    private var mUpdatedAt: String? = null

    @SerializedName("scopes")
    private var mScopes: List<String>? = null

    @SerializedName("fingerprint")
    private var mFingerprintUrl: String? = null

    private class App {
        @SerializedName("name")
        private var mName: String? = null

        @SerializedName("url")
        private var mUrl: String? = null

        @SerializedName("client_id")
        private var mClientId: String? = null
    }
}
